components/mbedtls_cxx/mbedtls_wrap.cpp

```diff
@@ -1,23 +1,36 @@
 /*
- * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ * SPDX-FileCopyrightText: 2023-2024 Espressif Systems (Shanghai) CO LTD
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+#include <mbedtls/timing.h>
 #include "mbedtls/ctr_drbg.h"
 #include "mbedtls/ssl.h"
 #include "mbedtls_wrap.hpp"
 
-bool Tls::init(is_server server, do_verify verify)
+bool Tls::init(is_server server, do_verify verify, TlsConfig *config)
 {
     const char pers[] = "mbedtls_wrapper";
+    if (config == nullptr) {
+
+    }
+    is_server_ = server == is_server{true};
+    is_dtls_ = config ? config->is_dtls : false;
+    uint32_t timeout = config ? config->timeout : 0;
+    const_buf client_id = config ? config->client_id : const_buf{};
     mbedtls_entropy_init(&entropy_);
     mbedtls_ctr_drbg_seed(&ctr_drbg_, mbedtls_entropy_func, &entropy_, (const unsigned char *)pers, sizeof(pers));
-    int ret = mbedtls_ssl_config_defaults(&conf_, server == is_server{true} ? MBEDTLS_SSL_IS_SERVER : MBEDTLS_SSL_IS_CLIENT, MBEDTLS_SSL_TRANSPORT_STREAM, MBEDTLS_SSL_PRESET_DEFAULT);
+    int endpoint = server == is_server{true} ? MBEDTLS_SSL_IS_SERVER : MBEDTLS_SSL_IS_CLIENT;
+    int transport = is_dtls_ ? MBEDTLS_SSL_TRANSPORT_DATAGRAM : MBEDTLS_SSL_TRANSPORT_STREAM;
+    int ret = mbedtls_ssl_config_defaults(&conf_, endpoint, transport, MBEDTLS_SSL_PRESET_DEFAULT);
     if (ret) {
         print_error("mbedtls_ssl_config_defaults", ret);
         return false;
     }
     mbedtls_ssl_conf_rng(&conf_, mbedtls_ctr_drbg_random, &ctr_drbg_);
+    if (timeout) {
+        mbedtls_ssl_conf_read_timeout(&conf_, 10000);
```

Comment:
Is this better to clarify the magic timeout  = 10000 here?

Reply:
Thanks

Reply:
Thank you for the fix @david-cermak!